### PR TITLE
Constructors

### PR DESCRIPTION
**Pull request by cloudRoutine**
uses importing constructors instead

### CODE REVIEW

File: src/FSharpVSPowerTools.Logic.VS2015/RecordStubGeneratorSuggestedActionsSourceProvider.fs [hunk 1]
```diff
@@ -17,48 +17,36 @@ open FSharpVSPowerTools
...
-    [<Import; DefaultValue>]
-    val mutable ProjectFactory: ProjectFactory
-
-    [<Import; DefaultValue>]
-    val mutable EditorOptionsFactory: IEditorOptionsFactoryService
-
-    [<Import; DefaultValue>]
-    val mutable OpenDocumentsTracker: IOpenDocumentsTracker
+type RecordStubGeneratorSuggestedActionsSourceProvider [<ImportingConstructor>]
+   (FSharpVsLanguageService: VSLanguageService,
```
Comment (dungpa) [inline src/FSharpVSPowerTools.Logic.VS2015/RecordStubGeneratorSuggestedActionsSourceProvider.fs]: Could you rename these fields to use camelCase?

File: src/FSharpVSPowerTools.Logic.VS2015/ResolveUnopenedNamespaceSuggestedActionsSourceProvider.fs [hunk 2]
```diff
@@ -18,34 +18,26 @@ open Microsoft.VisualStudio.Imaging.Interop
...
-    [<Import(typeof<SVsServiceProvider>); DefaultValue>]
-    val mutable ServiceProvider: IServiceProvider
-
-    [<Import; DefaultValue>]
-    val mutable UndoHistoryRegistry: ITextUndoHistoryRegistry
-
-    [<Import; DefaultValue>]
-    val mutable ProjectFactory: ProjectFactory
+type ResolveUnopenedNamespaceSuggestedActionsSourceProvider [<ImportingConstructor>]
+   (FSharpVsLanguageService: VSLanguageService,
```
Comment (dungpa) [inline src/FSharpVSPowerTools.Logic.VS2015/ResolveUnopenedNamespaceSuggestedActionsSourceProvider.fs]: Same comment as above.

File: src/FSharpVSPowerTools.Logic.VS2015/UnionPatternMatchCaseGeneratorSuggestedActionsSourceProvider.fs [hunk 3]
```diff
@@ -17,44 +17,34 @@ open FSharpVSPowerTools
...
-    val mutable UndoHistoryRegistry: ITextUndoHistoryRegistry
-
-    [<Import; DefaultValue>]
-    val mutable ProjectFactory: ProjectFactory
-
-    [<Import; DefaultValue>]
-    val mutable OpenDocumentsTracker: IOpenDocumentsTracker
+type UnionPatternMatchCaseSuggestedActionsSourceProvider [<ImportingConstructor>]
+   (FSharpVsLanguageService: VSLanguageService,
+    TextDocumentFactoryService: ITextDocumentFactoryService,
```
Comment (dungpa) [inline src/FSharpVSPowerTools.Logic.VS2015/UnionPatternMatchCaseGeneratorSuggestedActionsSourceProvider.fs]: Same comment as above.

**Comment by dungpa** [comment]
Could you rebase onto master? There are quite many unnecessary merge commits.

**Comment by cloudRoutine** [comment]
how do i stop those from happening all the time? they're really annoying

**Comment by dungpa** [comment]
I think `git rebase master` and `git push -f constructors` should do it. If there is no git conflict, git rebase goes through successfully and merge commits disappear.

**Comment by cloudRoutine** [comment]
I mean when I regularly update my fork from fsprojects/master that's when they all happen in the first place. should I always be rebasing instead of pulling?

**Comment by dungpa** [comment]
Yes, rebase is often cleaner.

**Comment by cloudRoutine** [comment]
do I need to do anything else?

**Comment by dungpa** [comment]
The constructors arguments are still in PasCal case? Did you fix that?

**Comment by cloudRoutine** [comment]
nope, i'll get that

**Comment by dungpa** [comment]
:+1:

**Pull request merged**